### ISSUE TITLE
Watcher: per action's throttling with condition

### ISSUE THREAD

**Issue by urusha** [elastic/elasticsearch#27358]
It would be nice to have per action's throttling with condition in watchers. With this we could make problem-ok messaging with one watcher. Currently, watcher doesn't reset throttle_period after action's condition became 'false', so we need to create 2 watchers (for problem and for ok, with watcher-wide conditions).
Here is what we'd like to have (but it's not working like we want):
* send problem messages every hour
* send ok message only once after resolution
```
 "condition": {
    "always": {}
  },
  "actions": {
    "notify-slack-problem": {
      "throttle_period": "1h",
      "condition": {
        "compare": {
          "ctx.payload.hits.total": {
            "lte": 150
          }
        }
      },
      "slack": {
        "message": {
          "attachments": [
            {
              "color": "danger",
              "title": "PROBLEM",
              "text": "{{ctx.payload.hits.total}}"
            }
          ]
        }
      }
    },
    "notify-slack-ok": {
      "throttle_period": "3650d",
      "condition": {
        "compare": {
          "ctx.payload.hits.total": {
            "gt": 150
          }
        }
      },
      "slack": {
        "message": {
          "attachments": [
            {
              "color": "good",
              "title": "OK",
              "text": "{{ctx.payload.hits.total}}"
            }
          ]
        }
      }
    }
  }
```
After the first ok message, `notify-slack-ok` action is throttling, so we will not receive it before the end of `throttle_period`:
ok (sent every 10y) -> problem (sent every 1h) -> ok (will only be sent after 10y from the last ok) -> problem (will only be sent after 1h from the last problem)

We could implement action's condition option like `throttle_reset` (default - false)

**Comment by spinscale** [comment]
Thanks a lot for your input!

We are planning to have some sort of configurable execution state per watch (and thus also per action ,as it is user configurable) in the future. This would allow you to not only have on/off detection but also detect flapping watches as part of your execution and than act accordingly. This will also solve your use-case.

I will however close this issue in the Elasticsearch repo, as this repository is only for the development and discussion of the open source part of Elasticsearch. Feel free to post any suggestions in our [x-pack discussion forum](https://discuss.elastic.co/c/x-pack) though. Happy to get any input and have further discussions!
